src/main/java/com/scottlogic/UserPost.java

```diff
@@ -4,8 +4,10 @@
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
 
+import javax.validation.constraints.NotNull;
+
 public class UserPost {
-    private String author;
+    @NotNull private String author;
```

Comment:
I'm not sure if this is really the appropriate use case for `@NotNull`. Without running it through a validator, it's not actually guarding against creating a UserPost with a null author - it just throws an exception if we try putting it through `Collections.sort()`.

We can look a bit deeper into it, but maybe leave it out for now and update your sorter and tests on the assumption that any field could be null.

Reply:
Currently, if any names are null, while sorting I return null and display "Sorting cannot be done".
I'm currently writing test cases for date sorting, and that does not allow me to enter any non integer values in the input

Reply:
I think I would still expect to get a sorted list back, even if I provide it with some null authors. I'd probably expect all the null authors to be at the end of the list.